# Final project

## Inport tools

In [301]:
# package import
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set()

## Load datasets

In [406]:
df_athletes = pd.read_csv('data/athletes.csv')
df_athletes.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [407]:
df_countries = pd.read_csv('data/countries.csv')
df_countries.head()

Country                               Region  Population  \
0     Afghanistan         ASIA (EX. NEAR EAST)             31056997   
1         Albania   EASTERN EUROPE                          3581655   
2         Algeria   NORTHERN AFRICA                        32930091   
3  American Samoa   OCEANIA                                   57794   
4         Andorra   WESTERN EUROPE                            71201   

   Area (sq. mi.) Pop. Density (per sq. mi.) Coastline (coast/area ratio)  \
0          647500                       48,0                         0,00   
1           28748                      124,6                         1,26   
2         2381740                       13,8                         0,04   
3             199                      290,4                        58,29   
4             468                      152,1                         0,00   

  Net migration Infant mortality (per 1000 births)  GDP ($ per capita)  \
0         23,06                             163,07               700.0   
1         -4,93                              21,52              4500.0   
2         -0,39                                 31              6000.0   
3        -20,71                               9,27              8000.0   
4           6,6                               4,05             19000.0   

  Literacy (%) Phones (per 1000) Arable (%) Crops (%) Other (%) Climate  \
0         36,0               3,2      12,13      0,22     87,65       1   
1         86,5              71,2      21,09      4,42     74,49       3   
2         70,0              78,1       3,22      0,25     96,53       1   
3         97,0             259,5         10        15        75       2   
4        100,0             497,2       2,22         0     97,78       3   

  Birthrate Deathrate Agriculture Industry Service  
0      46,6     20,34        0,38     0,24    0,38  
1     15,11      5,22       0,232    0,188   0,579  
2     17,14      4,61       0,101      0,6   0,298  
3     22,46      3,27         NaN      NaN     NaN  
4      8,71      6,25         NaN      NaN     NaN

In [408]:
df_GDP = pd.read_csv('data/GDP.csv')
df_GDP.head()

Country Name Country Code                Indicator Name  Indicator Code  \
0        Aruba          ABW  GDP per capita (current US$)  NY.GDP.PCAP.CD   
1  Afghanistan          AFG  GDP per capita (current US$)  NY.GDP.PCAP.CD   
2       Angola          AGO  GDP per capita (current US$)  NY.GDP.PCAP.CD   
3      Albania          ALB  GDP per capita (current US$)  NY.GDP.PCAP.CD   
4      Andorra          AND  GDP per capita (current US$)  NY.GDP.PCAP.CD   

        1960       1961       1962       1963       1964        1965  ...  \
0        NaN        NaN        NaN        NaN        NaN         NaN  ...   
1  59.773194  59.860874  58.458015  78.706388  82.095231  101.108305  ...   
2        NaN        NaN        NaN        NaN        NaN         NaN  ...   
3        NaN        NaN        NaN        NaN        NaN         NaN  ...   
4        NaN        NaN        NaN        NaN        NaN         NaN  ...   

           2010          2011          2012          2013          2014  \
0  23512.602596  24985.993281  24713.698045  25025.099563  25533.569780   
1    543.303042    591.162346    641.872034    637.165044    613.856333   
2   3587.883798   4615.468028   5100.095808   5254.882338   5408.410496   
3   4094.358816   4437.177794   4247.614342   4413.082887   4578.667934   
4  39736.354063  41100.729938  38392.943901  40626.751632  42300.334128   

           2015          2016          2017          2018  2019  
0  25796.380251  25239.600411  25630.266492           NaN   NaN  
1    578.466353    547.228110    556.302139    520.896603   NaN  
2   4166.979684   3506.072885   4095.812942   3432.385736   NaN  
3   3952.830781   4124.108543   4532.889198   5253.630064   NaN  
4  36039.653496  37224.108916  39134.393371  42029.762737   NaN  

[5 rows x 64 columns]

## Manage NaN values and adapt it for analysis

#### Athletes dataset

In [409]:
df_athletes = df_athletes.drop(columns=["Name","ID","Games","Event"])
df_athletes.drop( df_athletes[ df_athletes['Year'] < 1960].index , inplace=True) # the data we are droping because of the countries dataset

In [410]:
clust_a=df_athletes
df_athletes["Medal"]=df_athletes.Medal.map({"Gold":3, "Silver":2,"Bronze":1})
df_athletes=df_athletes.fillna(0)
df_athletes.head()

Sex   Age  Height  Weight         Team  NOC  Year  Season         City  \
0   M  24.0   180.0    80.0        China  CHN  1992  Summer    Barcelona   
1   M  23.0   170.0    60.0        China  CHN  2012  Summer       London   
4   F  21.0   185.0    82.0  Netherlands  NED  1988  Winter      Calgary   
5   F  21.0   185.0    82.0  Netherlands  NED  1988  Winter      Calgary   
6   F  25.0   185.0    82.0  Netherlands  NED  1992  Winter  Albertville   

           Sport  Medal  
0     Basketball    0.0  
1           Judo    0.0  
4  Speed Skating    0.0  
5  Speed Skating    0.0  
6  Speed Skating    0.0

##### 0 = No medals won
##### 1 = Bronze medal won
##### 2 = Silver medal won
##### 3 = Golden medal won


In [307]:
df_athletes=df_athletes.sort_values(by='Team') # Put it in alphabetical order to see clearer
df_athletes.head()

Sex   Age  Height  Weight         Team  NOC  Year  Season     City  \
193165   M  52.0   180.0    84.0    Acipactli  MEX  1964  Summer    Tokyo   
73071    M  46.0   173.0    73.0    Acipactli  MEX  1964  Summer    Tokyo   
51134    M  44.0   171.0    69.0    Acipactli  MEX  1964  Summer    Tokyo   
12535    M  28.0   186.0    68.0  Afghanistan  AFG  2012  Summer   London   
264047   F  22.0   180.0    56.0  Afghanistan  AFG  2008  Summer  Beijing   

            Sport  Medal  
193165    Sailing    0.0  
73071     Sailing    0.0  
51134     Sailing    0.0  
12535   Taekwondo    0.0  
264047  Athletics    0.0

In [308]:
#Check for NaN values
df_athletes.isna().sum()

Sex       0
Age       0
Height    0
Weight    0
Team      0
NOC       0
Year      0
Season    0
City      0
Sport     0
Medal     0
dtype: int64

There are no longer NaN values, we can continue

In [309]:
#df_summer = df_athletes[df_athletes['Season']=='Summer']
#df_summer.head()

In [310]:
#df_winter = df_athletes[df_athletes['Season']=='Winter']
#df_winter.head()

#### Countries dataset

In [311]:
display(df_countries.isna().sum(),df_countries.count())

Country                                0
Region                                 0
Population                             0
Area (sq. mi.)                         0
Pop. Density (per sq. mi.)             0
Coastline (coast/area ratio)           0
Net migration                          3
Infant mortality (per 1000 births)     3
GDP ($ per capita)                     1
Literacy (%)                          18
Phones (per 1000)                      4
Arable (%)                             2
Crops (%)                              2
Other (%)                              2
Climate                               22
Birthrate                              3
Deathrate                              4
Agriculture                           15
Industry                              16
Service                               15
dtype: int64

Country                               227
Region                                227
Population                            227
Area (sq. mi.)                        227
Pop. Density (per sq. mi.)            227
Coastline (coast/area ratio)          227
Net migration                         224
Infant mortality (per 1000 births)    224
GDP ($ per capita)                    226
Literacy (%)                          209
Phones (per 1000)                     223
Arable (%)                            225
Crops (%)                             225
Other (%)                             225
Climate                               205
Birthrate                             224
Deathrate                             223
Agriculture                           212
Industry                              211
Service                               212
dtype: int64

##### 22<<<227 so we might just drop those Na values

In [312]:
df_countries=df_countries.dropna()

In [313]:
df_countries.isna().sum().sum()
clust=df_countries

No Na values in the dataset. We can continue

#### GDP dataset

In [314]:
#GDP per capita in US $ during the years

In [315]:
df_GDP=df_GDP.drop(columns=["Country Name","Indicator Name","Indicator Code","2018","2019"]) # the last games were in 2016 so we don't need it

In [316]:
df_GDP.head()

Country Code       1960       1961       1962       1963       1964  \
0          ABW        NaN        NaN        NaN        NaN        NaN   
1          AFG  59.773194  59.860874  58.458015  78.706388  82.095231   
2          AGO        NaN        NaN        NaN        NaN        NaN   
3          ALB        NaN        NaN        NaN        NaN        NaN   
4          AND        NaN        NaN        NaN        NaN        NaN   

         1965        1966        1967        1968  ...          2008  \
0         NaN         NaN         NaN         NaN  ...  27084.703690   
1  101.108305  137.594352  160.898589  129.108323  ...    364.660465   
2         NaN         NaN         NaN         NaN  ...   4080.941410   
3         NaN         NaN         NaN         NaN  ...   4370.539925   
4         NaN         NaN         NaN         NaN  ...  47785.089273   

           2009          2010          2011          2012          2013  \
0  24630.453714  23512.602596  24985.993281  24713.698045  25025.099563   
1    438.076034    543.303042    591.162346    641.872034    637.165044   
2   3122.780766   3587.883798   4615.468028   5100.095808   5254.882338   
3   4114.134899   4094.358816   4437.177794   4247.614342   4413.082887   
4  43338.866758  39736.354063  41100.729938  38392.943901  40626.751632   

           2014          2015          2016          2017  
0  25533.569780  25796.380251  25239.600411  25630.266492  
1    613.856333    578.466353    547.228110    556.302139  
2   5408.410496   4166.979684   3506.072885   4095.812942  
3   4578.667934   3952.830781   4124.108543   4532.889198  
4  42300.334128  36039.653496  37224.108916  39134.393371  

[5 rows x 59 columns]

In [317]:
df_GDP.describe()

1960         1961         1962         1963         1964  \
count   133.000000   134.000000   136.000000   136.000000   136.000000   
mean    477.090548   510.997180   537.759283   570.571590   620.118759   
std     597.614845   642.387548   674.781661   714.643153   782.415979   
min      40.537115    40.689390    34.790581    40.752237    41.083814   
25%     107.949993   109.211954   114.955692   123.204376   124.980086   
50%     231.025649   217.409205   232.156045   252.392332   259.622516   
75%     489.351689   561.473554   634.966539   666.844461   752.171293   
max    3007.123445  3066.562869  3243.843078  3374.515171  3573.941185   

              1965         1966         1967         1968         1969  ...  \
count   146.000000   147.000000   150.000000   154.000000   154.000000  ...   
mean    687.168945   730.812250   747.106490   766.201365   828.090061  ...   
std     874.030595   940.758849   971.350324  1002.331545  1076.088550  ...   
min      45.989354    37.488783    46.642770    48.784045    51.794609  ...   
25%     141.029619   145.767969   157.983788   159.262062   161.937705  ...   
50%     273.655226   277.233872   278.460057   291.141585   305.373736  ...   
75%     854.120036   882.906899   875.547972   805.809962   860.595621  ...   
max    4443.405272  4571.181955  4336.426587  4695.923390  5032.144743  ...   

                2008           2009           2010           2011  \
count     251.000000     251.000000     251.000000     252.000000   
mean    15391.724830   13707.443017   14423.626372   15903.921344   
std     24146.891838   21083.697615   21710.830053   23961.297976   
min       198.352901     212.136880     234.235647     249.577979   
25%      1470.043561    1357.187492    1575.699381    1738.219933   
50%      5090.932345    4714.702495    5324.617040    5968.327288   
75%     18372.453584   15860.284016   16328.315060   18398.905381   
max    185721.794154  154762.199427  150585.448911  168785.940809   

                2012           2013           2014           2015  \
count     251.000000     252.000000     251.000000     250.000000   
mean    15822.664863   16354.589444   16356.195139   14707.131596   
std     23150.370578   24726.589825   25092.278876   22369.580670   
min       252.358980     256.976003     248.844951     293.465423   
25%      1873.610010    1957.578334    2035.213733    2009.418262   
50%      6528.972218    6792.886617    6600.056650    6036.798975   
75%     18291.925068   19283.183735   18999.005287   16666.300061   
max    157515.899069  177593.351895  189170.895671  167290.939984   

                2016           2017  
count     250.000000     246.000000  
mean    14747.116816   14515.124497  
std     22472.843662   20608.291936  
min       282.149682     293.009062  
25%      2050.138368    2006.922358  
50%      5813.802359    6170.481787  
75%     16998.588687   16409.835459  
max    169904.182096  166726.051926  

[8 rows x 58 columns]

In [318]:
# Getting the min value of the last 4 years for the current year to fill the NaN values

#df_GDP = df_GDP.fillna(method='backfill', inplace = True, axis=1)
df_GDP = df_GDP.T
df_GDP.head()

0        1    2    3    4    5    6        7    8    9    ...  \
Country Code  ABW      AFG  AGO  ALB  AND  ARB  ARE      ARG  ARM  ASM  ...   
1960          NaN  59.7732  NaN  NaN  NaN  NaN  NaN      NaN  NaN  NaN  ...   
1961          NaN  59.8609  NaN  NaN  NaN  NaN  NaN      NaN  NaN  NaN  ...   
1962          NaN   58.458  NaN  NaN  NaN  NaN  NaN  1155.89  NaN  NaN  ...   
1963          NaN  78.7064  NaN  NaN  NaN  NaN  NaN  850.305  NaN  NaN  ...   

              254  255  256      257  258  259  260      261      262      263  
Country Code  VIR  VNM  VUT      WLD  WSM  XKX  YEM      ZAF      ZMB      ZWE  
1960          NaN  NaN  NaN  452.078  NaN  NaN  NaN   443.01  232.189  278.814  
1961          NaN  NaN  NaN  464.739  NaN  NaN  NaN  454.962  220.042  280.829  
1962          NaN  NaN  NaN  490.507  NaN  NaN  NaN  473.012  212.578  276.689  
1963          NaN  NaN  NaN  517.419  NaN  NaN  NaN  511.497  213.896   277.48  

[5 rows x 264 columns]

In [319]:
df_GDP = df_GDP.fillna(method='backfill', axis = 0)
df_GDP = df_GDP.fillna(method='ffill', axis = 0)

In [320]:
df_GDP = df_GDP.T

In [321]:
df_GDP.head(5)

Country Code     1960     1961     1962     1963     1964     1965     1966  \
0          ABW   6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   
1          AFG  59.7732  59.8609   58.458  78.7064  82.0952  101.108  137.594   
2          AGO  710.982  710.982  710.982  710.982  710.982  710.982  710.982   
3          ALB  639.485  639.485  639.485  639.485  639.485  639.485  639.485   
4          AND  3238.56  3238.56  3238.56  3238.56  3238.56  3238.56  3238.56   

      1967     1968  ...     2008     2009     2010     2011     2012  \
0   6472.5   6472.5  ...  27084.7  24630.5  23512.6    24986  24713.7   
1  160.899  129.108  ...   364.66  438.076  543.303  591.162  641.872   
2  710.982  710.982  ...  4080.94  3122.78  3587.88  4615.47   5100.1   
3  639.485  639.485  ...  4370.54  4114.13  4094.36  4437.18  4247.61   
4  3238.56  3238.56  ...  47785.1  43338.9  39736.4  41100.7  38392.9   

      2013     2014     2015     2016     2017  
0  25025.1  25533.6  25796.4  25239.6  25630.3  
1  637.165  613.856  578.466  547.228  556.302  
2  5254.88  5408.41  4166.98  3506.07  4095.81  
3  4413.08  4578.67  3952.83  4124.11  4532.89  
4  40626.8  42300.3  36039.7  37224.1  39134.4  

[5 rows x 59 columns]

In [322]:
df_GDP.describe()

Country Code        1960        1961        1962        1963  \
count           264  264.000000  264.000000  264.000000  264.000000   
unique          264  262.000000  262.000000  262.000000  262.000000   
top             AUS  131.654994   85.945754   89.792026   98.841369   
freq              1    2.000000    2.000000    2.000000    2.000000   

              1964        1965        1966        1967        1968  ...  \
count   264.000000  264.000000  264.000000  264.000000  264.000000  ...   
unique  262.000000  262.000000  262.000000  262.000000  262.000000  ...   
top     150.828798  163.896432  172.554294  166.347815  173.469143  ...   
freq      2.000000    2.000000    2.000000    2.000000    2.000000  ...   

               2008         2009         2010         2011         2012  \
count    264.000000   264.000000   264.000000   264.000000   264.000000   
unique   262.000000   262.000000   262.000000   262.000000   262.000000   
top     1470.043561  1357.618445  1576.176004  1367.338787  1364.441602   
freq       2.000000     2.000000     2.000000     2.000000     2.000000   

               2013         2014         2015       2016         2017  
count    264.000000   264.000000   264.000000   264.0000   264.000000  
unique   262.000000   262.000000   263.000000   262.0000   262.000000  
top     1844.577249  1885.558172  1541.835813  1511.3049  1599.336187  
freq       2.000000     2.000000     2.000000     2.0000     2.000000  

[4 rows x 59 columns]

In [323]:
df_GDP_4years = df_GDP

In [324]:
df_GDP_4years.set_index('Country Code', inplace=True)
df_GDP_4years.head()

1960     1961     1962     1963     1964     1965     1966  \
Country Code                                                                  
ABW            6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   
AFG           59.7732  59.8609   58.458  78.7064  82.0952  101.108  137.594   
AGO           710.982  710.982  710.982  710.982  710.982  710.982  710.982   
ALB           639.485  639.485  639.485  639.485  639.485  639.485  639.485   
AND           3238.56  3238.56  3238.56  3238.56  3238.56  3238.56  3238.56   

                 1967     1968     1969  ...     2008     2009     2010  \
Country Code                             ...                              
ABW            6472.5   6472.5   6472.5  ...  27084.7  24630.5  23512.6   
AFG           160.899  129.108   129.33  ...   364.66  438.076  543.303   
AGO           710.982  710.982  710.982  ...  4080.94  3122.78  3587.88   
ALB           639.485  639.485  639.485  ...  4370.54  4114.13  4094.36   
AND           3238.56  3238.56  3238.56  ...  47785.1  43338.9  39736.4   

                 2011     2012     2013     2014     2015     2016     2017  
Country Code                                                                 
ABW             24986  24713.7  25025.1  25533.6  25796.4  25239.6  25630.3  
AFG           591.162  641.872  637.165  613.856  578.466  547.228  556.302  
AGO           4615.47   5100.1  5254.88  5408.41  4166.98  3506.07  4095.81  
ALB           4437.18  4247.61  4413.08  4578.67  3952.83  4124.11  4532.89  
AND           41100.7  38392.9  40626.8  42300.3  36039.7  37224.1  39134.4  

[5 rows x 58 columns]

In [325]:
#def get_min_4years(x):
    #if(x[])
    #return x

In [326]:
i = 1960
a = {}
while i < 2019:
    a[str(i)] = [str(i-1), str(i-2), str(i-3), str(i-4)]
    i+=1

In [327]:
#df_GDP_4years[a.get('1964')].min(axis=1)
#df_GDP_4years[a[str('1964')]]

In [328]:
df_GDP_new = df_GDP_4years.head()

In [329]:
df_GDP_new

1960     1961     1962     1963     1964     1965     1966  \
Country Code                                                                  
ABW            6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   
AFG           59.7732  59.8609   58.458  78.7064  82.0952  101.108  137.594   
AGO           710.982  710.982  710.982  710.982  710.982  710.982  710.982   
ALB           639.485  639.485  639.485  639.485  639.485  639.485  639.485   
AND           3238.56  3238.56  3238.56  3238.56  3238.56  3238.56  3238.56   

                 1967     1968     1969  ...     2008     2009     2010  \
Country Code                             ...                              
ABW            6472.5   6472.5   6472.5  ...  27084.7  24630.5  23512.6   
AFG           160.899  129.108   129.33  ...   364.66  438.076  543.303   
AGO           710.982  710.982  710.982  ...  4080.94  3122.78  3587.88   
ALB           639.485  639.485  639.485  ...  4370.54  4114.13  4094.36   
AND           3238.56  3238.56  3238.56  ...  47785.1  43338.9  39736.4   

                 2011     2012     2013     2014     2015     2016     2017  
Country Code                                                                 
ABW             24986  24713.7  25025.1  25533.6  25796.4  25239.6  25630.3  
AFG           591.162  641.872  637.165  613.856  578.466  547.228  556.302  
AGO           4615.47   5100.1  5254.88  5408.41  4166.98  3506.07  4095.81  
ALB           4437.18  4247.61  4413.08  4578.67  3952.83  4124.11  4532.89  
AND           41100.7  38392.9  40626.8  42300.3  36039.7  37224.1  39134.4  

[5 rows x 58 columns]

In [330]:
#df_GDP_new[df_GDP_new.index == 'ABW']

In [331]:
abw = df_GDP_new[a.get(str(1964))].min(axis=1)[df_GDP_new.index == 'ABW']
print(abw)
abw+20

Country Code
ABW    6472.502029
dtype: float64


Country Code
ABW    6492.502029
dtype: float64

In [332]:
for i, country in df_GDP_new.iterrows(): 
    #print(i, year) 
    #print() 
    year = 2017
    #print("------------- ", i, "---------------")
    while(year > 1963):
        #print("-----------------")
        #print(country[str(year)], " = ", df_GDP_new[a.get(str(year))].min(axis=1)[df_GDP_new.index == str(i)])
        country[str(year)] = float(df_GDP_new[a.get(str(year))].min(axis=1)[df_GDP_new.index == str(i)])
        #print("-----------------")
        year -= 1

#idx = 2017
#for index, row in df_GDP_new.iterrows():
    #print("--------- new row---------")    
    #print("index: ", idx, "\n", df_GDP_new[a.get(str(idx))].min(axis=1))
    #idx += -1
    #print(df_GDP_new[a.get(str(idx))].min(axis=1))
            #df_GDP_new.index(row.index()) = row
            #row[str(idx)] = df_GDP_new[a.get(str(idx))].min(axis=1)

In [333]:
df_GDP_new

1960     1961     1962     1963     1964     1965     1966  \
Country Code                                                                  
ABW            6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   6472.5   
AFG           59.7732  59.8609   58.458  78.7064   58.458   58.458   58.458   
AGO           710.982  710.982  710.982  710.982  710.982  710.982  710.982   
ALB           639.485  639.485  639.485  639.485  639.485  639.485  639.485   
AND           3238.56  3238.56  3238.56  3238.56  3238.56  3238.56  3238.56   

                 1967     1968     1969  ...     2008     2009     2010  \
Country Code                             ...                              
ABW            6472.5   6472.5   6472.5  ...    22570    23300  24045.3   
AFG           78.7064  82.0952  101.108  ...  211.382  242.031  263.734   
AGO           710.982  710.982  710.982  ...  1255.56  1902.42  2599.57   
ALB           639.485  639.485  639.485  ...  2373.58  2673.79  2972.74   
AND           3238.56  3238.56  3238.56  ...  38503.5    41282  43338.9   

                 2011     2012     2013     2014     2015     2016     2017  
Country Code                                                                 
ABW           23512.6  23512.6  23512.6  23512.6  24713.7  24713.7  25025.1  
AFG           359.693   364.66  438.076  543.303  591.162  578.466  547.228  
AGO              3122  3122.78  3122.78  3587.88  4615.47  4166.98  3506.07  
ALB           3595.04  4094.36  4094.36  4094.36  4247.61  3952.83  3952.83  
AND           39736.4  39736.4  38392.9  38392.9  38392.9  36039.7  36039.7  

[5 rows x 58 columns]

In [334]:
#print(df_GDP_4years.iloc[249])

In [335]:
#df_at = df_athletes
#n = df_GDP_4years[df_GDP_4years['Country Code'] == 'ALB']['1960']
#n

In [336]:
#gdp = []

In [337]:
#df_GDP_4years['Country Code']
#i = 0
#for index, row in df_at.iterrows():
    #i+= 1
    ##gdp.append(df_GDP_4years[df_GDP_4years['Country Code'] == row['NOC']][row['Year']])

In [338]:
#i

In [339]:
#ex = df_at.head(20)
#ex

In [340]:
#i = 0
#ex2 = []
#for index, e in ex.iterrows():
    # get country
    #country = df_GDP_4years[df_GDP_4years['Country Code'] == e['NOC']]
    #year = str(e['Year'])
    
    #print("--------------------\n", year, "\n", country[year])

In [341]:
#i = 0
#ex2 = []
#for index, e in ex.iterrows():
    # get country
    #country_idx = df_GDP_4years[df_GDP_4years['Country Code'] == e['NOC']].index
    # get GDP of year 
    #print(str(e['Year']))
    #gdp = df_GDP_4years.iloc[country_idx][str(e['Year'])]
    #print("\n--------\n gdp:\n", gdp, "\n-------------")
    #gdp = 
    #print(e['NOC'])
    #print(e['Year'])
    #print(e['Year'])
    #year = e['Year']
    #print(df_GDP_4years.loc[1])
    #df_GDP_4years[str(e['Year'])]
    #print(df_GDP_4years[e['Year']])
    #ex2.append(df_GDP_4years[df_GDP_4years['Country Code'] == e['NOC']]['2018'])

In [342]:
#df = pd.DataFrame(ex2) 
#df

In [343]:
#df_at['GDP'] = df_GDP_4years.loc[df_GDP_4years['Country Code'] == df_at['NOC']]['1960']

## Merge first two datasets

In [344]:
df_athletes.loc[df_athletes.NOC == 'NED', 'NOC'] = 'NER'
df_athletes = df_athletes.rename(columns={'Team': 'Country'})
df_athletes=df_athletes.set_index(df_athletes["Country"]).transpose()
df_countries=df_countries.set_index(df_countries["Country"]).transpose()
display(df_athletes.head(),df_countries.head())

Country  Acipactli  Acipactli  Acipactli  Afghanistan  Afghanistan  \
Sex              M          M          M            M            F   
Age             52         46         44           28           22   
Height         180        173        171          186          180   
Weight          84         73         69           68           56   
Country  Acipactli  Acipactli  Acipactli  Afghanistan  Afghanistan   

Country  Afghanistan  Afghanistan  Afghanistan  Afghanistan  Afghanistan  ...  \
Sex                M            M            M            M            M  ...   
Age                0           24           20           27           23  ...   
Height             0          186          179          168          168  ...   
Weight             0           68           75           64           64  ...   
Country  Afghanistan  Afghanistan  Afghanistan  Afghanistan  Afghanistan  ...   

Country  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  \
Sex             M         M         M         M         M         M         F   
Age            24        24        24        31        21        21        35   
Height        190       190       190       186       180       180         0   
Weight         82        82        82        70        60        60         0   
Country  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe   

Country  Zimbabwe  Zimbabwe  Zimbabwe  
Sex             M         M         M  
Age            25        20        41  
Height        180       170       175  
Weight         60        75        74  
Country  Zimbabwe  Zimbabwe  Zimbabwe  

[5 rows x 209072 columns]

Country                                      Afghanistan   \
Country                                      Afghanistan    
Region                      ASIA (EX. NEAR EAST)            
Population                                       31056997   
Area (sq. mi.)                                     647500   
Pop. Density (per sq. mi.)                           48,0   

Country                                                Albania   \
Country                                                Albania    
Region                      EASTERN EUROPE                        
Population                                              3581655   
Area (sq. mi.)                                            28748   
Pop. Density (per sq. mi.)                                124,6   

Country                                                Algeria   \
Country                                                Algeria    
Region                      NORTHERN AFRICA                       
Population                                             32930091   
Area (sq. mi.)                                          2381740   
Pop. Density (per sq. mi.)                                 13,8   

Country                                   Anguilla        Antigua & Barbuda   \
Country                                   Anguilla        Antigua & Barbuda    
Region                      LATIN AMER. & CARIB      LATIN AMER. & CARIB       
Population                                    13477                    69108   
Area (sq. mi.)                                  102                      443   
Pop. Density (per sq. mi.)                    132,1                    156,0   

Country                                  Argentina               Armenia   \
Country                                  Argentina               Armenia    
Region                      LATIN AMER. & CARIB      C.W. OF IND. STATES    
Population                                 39921833               2976372   
Area (sq. mi.)                              2766890                 29800   
Pop. Density (per sq. mi.)                     14,4                  99,9   

Country                                      Aruba   \
Country                                      Aruba    
Region                      LATIN AMER. & CARIB       
Population                                    71891   
Area (sq. mi.)                                  193   
Pop. Density (per sq. mi.)                    372,5   

Country                                              Australia   \
Country                                              Australia    
Region                      OCEANIA                               
Population                                             20264082   
Area (sq. mi.)                                          7686850   
Pop. Density (per sq. mi.)                                  2,6   

Country                                                Austria   ...  \
Country                                                Austria   ...   
Region                      WESTERN EUROPE                       ...   
Population                                              8192880  ...   
Area (sq. mi.)                                            83870  ...   
Pop. Density (per sq. mi.)                                 97,7  ...   

Country                                         United Kingdom   \
Country                                         United Kingdom    
Region                      WESTERN EUROPE                        
Population                                             60609153   
Area (sq. mi.)                                           244820   
Pop. Density (per sq. mi.)                                247,6   

Country                                          United States   \
Country                                          United States    
Region                      NORTHERN AMERICA                      
Population                                            298444215   
Area (sq. mi.)                               

# Clustering

### As Clustering is one type of unsupervised learning, due to We don't given any labels associated with the data, We will try to make a data analisys for compering how the industry level of each country is related to the number of won medal in Olimpyc games.

In [413]:
aux1=clust_a
clust_a.head()

Sex   Age  Height  Weight         Team  NOC  Year  Season         City  \
0   M  24.0   180.0    80.0        China  CHN  1992  Summer    Barcelona   
1   M  23.0   170.0    60.0        China  CHN  2012  Summer       London   
4   F  21.0   185.0    82.0  Netherlands  NED  1988  Winter      Calgary   
5   F  21.0   185.0    82.0  Netherlands  NED  1988  Winter      Calgary   
6   F  25.0   185.0    82.0  Netherlands  NED  1992  Winter  Albertville   

           Sport  Medal  
0     Basketball    NaN  
1           Judo    NaN  
4  Speed Skating    NaN  
5  Speed Skating    NaN  
6  Speed Skating    NaN

In [414]:
aux1['Medal_Won'] = aux1.Medal.map({2:1, 3:1, 1:1, np.nan:0})

In [418]:
#aux1.head()
aux2=aux1[aux1.Year == 2016]
aux2[aux2.Medal_Won == 1].head()

Sex   Age  Height  Weight        Team  NOC  Year  Season            City  \
158   M  21.0   198.0    90.0       Italy  ITA  2016  Summer  Rio de Janeiro   
161   F  21.0   165.0    49.0  Azerbaijan  AZE  2016  Summer  Rio de Janeiro   
175   M  31.0   182.0    86.0      France  FRA  2016  Summer  Rio de Janeiro   
450   M  26.0   170.0    80.0        Iran  IRI  2016  Summer  Rio de Janeiro   
794   M  24.0   161.0    62.0      Russia  RUS  2016  Summer  Rio de Janeiro   

          Sport  Medal  Medal_Won  
158      Rowing    1.0          1  
161   Taekwondo    1.0          1  
175    Handball    2.0          1  
450   Wrestling    1.0          1  
794  Gymnastics    2.0          1

In [417]:
medalC = aux2.groupby(['Team'])['Medal_Won'].agg('sum').reset_index()
medal_count = medalC.sort_values('Medal_Won',ascending = False)
#medal_count_per_country = medal_count_per_country.iloc[:136,:]
medal_count.head()

Team  Medal_Won
237  United States        256
87         Germany        157
91   Great Britain        145
184         Russia        113
50           China        109

In [403]:
aux1=aux1.drop(columns=["Medal_Won"])
clust_a=clust_a.drop(columns=["Medal_Won"])

In [390]:
aux1.head()

Sex   Age  Height  Weight         Team  NOC  Year  Season         City  \
0   M  24.0   180.0    80.0        China  CHN  1992  Summer    Barcelona   
1   M  23.0   170.0    60.0        China  CHN  2012  Summer       London   
4   F  21.0   185.0    82.0  Netherlands  NED  1988  Winter      Calgary   
5   F  21.0   185.0    82.0  Netherlands  NED  1988  Winter      Calgary   
6   F  25.0   185.0    82.0  Netherlands  NED  1992  Winter  Albertville   

           Sport  Medal  
0     Basketball    NaN  
1           Judo    NaN  
4  Speed Skating    NaN  
5  Speed Skating    NaN  
6  Speed Skating    NaN